core/src/main/java/org/infinispan/transaction/xa/XaTransactionTable.java

```diff
@@ -180,7 +181,7 @@ private void forgetSuccessfullyCompletedTransaction(RecoveryManager recoveryMana
          LocalXaTransaction localTransaction, boolean committedInOnePhase) {
       final GlobalTransaction gtx = localTransaction.getGlobalTransaction();
       if (isRecoveryEnabled()) {
-         recoveryManager.removeRecoveryInformation(localTransaction.getRemoteLocksAcquired(), xid, false, gtx,
+         recoveryManager.removeRecoveryInformation(localTransaction.getRemoteLocksAcquired(), xid, gtx,
```

Comment:
It helps tx performance if we can tell the transaction manager that we've committed without waiting for another ack from the remote nodes (we already wait for an ack for the prepare and optionally commit).

Reply:
Not sure I follow. The sync parameter just handled if it should block waiting on the call https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/transaction/xa/recovery/RecoveryManagerImpl.java#L159 Now instead of doing that we return a `CompletionStage` and the places that were passing false do not check it, which means it won't wait.

Reply:
My mistake, I saw a `return` keyword that wasn't there :)

There's still one extra response message to send/receive with `invokeCommand` vs. `sendTo`, but it's not that much.